lib/spack/spack/spec.py

```diff
@@ -204,6 +204,14 @@
 specfile_format_version = 3
 
 
+class InstallStatus:
+    installed = 0
+    upstream = 1
+    missing = 2
+    absent = 3
+    external = 4
```

Comment:
I think it's fine, but maybe we should leave a comment that this is purely for display purposes, since statuses are not mutually exclusive (you can have missing, installed, upstream and absent externals)

Reply:
Did you see the `purple` example at https://docs.python.org/3/library/enum.html#enum.Flag.__contains__?

Reply:
Yeah, flags are better than enums for that. But if the use case is really about showing a status symbol and we can only show one symbol, i guess the enum is OK.

Reply:
Can we use `Enum` consistently for new code, since we support running on 3.6+ ?